crates/gosub_pipeline/tools/souper/soupertoo.py

```
@@ -0,0 +1,84 @@
+async def main():
+    if len(sys.argv) != 2:
+        print("Usage: souper.py <url>")
+        sys.exit(1)
+
+    dom_tree = await fetch_and_parse_html(sys.argv[1])
+
+    # Save to JSON file
+    with open("output.json", "w", encoding="utf-8") as f:
+        json.dump(dom_tree, f, indent=2, ensure_ascii=False)
+
+    print(f"DOM tree with computed styles saved to output.json")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)